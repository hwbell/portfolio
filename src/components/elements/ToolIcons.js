import React from "react";

// element data
import {toolsImages} from '../../assets/toolsImages';

class ToolIcons extends React.Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  renderTools() {
    let imageId;
    return this.props.tools.map((tool, i) => {
      imageId = `${this.props.job.split(" ").join("")}-tool${i}`;

      return (
        <div
          id={imageId}
          className="tool-image col"
          key={i}
          style={styles.toolHolder}
        >
          <a
            target="_blank"
            href={`https://www.google.com/search?q=${tool}`}
            alt={`link to ${tool}`}
          >
            <img
              // onMouseOver={() => {
              //   this.togglePopover(tool);
              // }}
              // onMouseLeave={() => {
              //   this.togglePopover();
              // }}
              style={styles.toolImage}
              src={toolsImages[tool]}
              alt="no image"
            />
          </a>
        </div>
      );
    });
  }

  render() {
    let imageId;

    return (
      <div className="tool-icons">
        {this.props.tools.map((tool, i) => {
          imageId = `${this.props.job.split(" ").join("")}-tool${i}`;

          return (
            <div
              id={imageId}
              className="tool-image col"
              key={i}
              style={styles.toolHolder}
            >
              <a
                target="_blank"
                href={`https://www.google.com/search?q=${tool}`}
                alt={`link to ${tool}`}
              >
                <img
                  // onMouseOver={() => {
                  //   this.togglePopover(tool);
                  // }}
                  // onMouseLeave={() => {
                  //   this.togglePopover();
                  // }}
                  style={styles.toolImage}
                  src={toolsImages[tool]}
                  alt="no image"
                />
              </a>
            </div>
          );
        })}
      </div>
    );
  }
}

const styles = {
  toolHolder: {
    maxWidth: "50px",
    backgroundColor: "white",
  },
  toolImage: {
    height: "35px",
    width: "35px",
  },
};

export default ToolIcons;
