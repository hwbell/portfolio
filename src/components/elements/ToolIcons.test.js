import React from 'react';
import ReactDOM from 'react-dom';
import ToolIcons from './ToolIcons';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

// react router context
// import { MemoryRouter } from 'react-router-dom';

describe('ToolIcons Component', () => {

  let wrapper;
  let someProps;
  beforeEach(() => {
    someProps = {
      year: "2010",
      job: "Research Associate at Brandeis University",
      tools: [ "matlab", "dna", "microscope"],
      logo: require("../../assets/icons/brandeis.png"),
      modal: {
        title: "Research Associate at Brandeis University",
        bullets: [
          'Developed image analysis software tools to streamline collection and processing of experimental data.',
          'Managed development of experimental methods and data analysis.',
          'Managed daily operations of research laboratory.',
        ],
        description: `At Brandeis University, I was an integral part of a team of researchers dissecting the cellular and genetic mechanisms of neuronal communication and memory storage. During my time in Dr. Pial Sengupta's laboratory, I was a leading member of the 'Thermotaxis' group, which studied temperature sensation in individual neurons as well as networks. I developed many analysis tools with the technical computing software MATLAB to help the team analyze data more efficiently and accurately. I also managed the day to day operations of the lab.` 
      }
    }

    wrapper = shallow(
      // we only need the MemoryRouter if there are links to routes of the react-router(pages) in the component that we need to test
      // <MemoryRouter>
      <ToolIcons {...someProps} />
      // </MemoryRouter>
    );

  })

  // tests
  it('renders without crashing', async () => {
    wrapper.update();
  });

  it('renders correctly', async () => {
    const tree = await renderer
      .create(<ToolIcons {...someProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains the correct elements for different data', () => {
    expect(wrapper.find('.tool-icons').length).toEqual(1);
    expect(wrapper.find('.tool-image').length).toEqual(3);
  })

  // it('calls click event', () => {
  //   const FakeFun = jest.spyOn(ToolIcons.prototype, 'renderIcons');
  //   const newWrapper = shallow((<ToolIcons />));
  //   newWrapper.update();
  //   expect(FakeFun).toHaveBeenCalled();
  // });

  // it('should render all buttons with info from provided props.buttons', () => {
  //   expect(wrapper.find('Link').at(0).props().to).toBe(someProps.buttons[0].link)
  //   expect(wrapper.find('Link').at(1).props().to).toBe(someProps.buttons[1].link)
  //   expect(wrapper.find('Link').at(2).props().to).toBe(someProps.buttons[2].link)

  // })
})