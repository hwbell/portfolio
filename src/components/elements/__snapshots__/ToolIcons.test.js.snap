// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolIcons Component renders correctly 1`] = `
<div
  className="tool-icons"
>
  <div
    className="tool-image col"
    id="ResearchAssociateatBrandeisUniversity-tool0"
    style={
      Object {
        "backgroundColor": "white",
        "maxWidth": "50px",
      }
    }
  >
    <a
      alt="link to matlab"
      href="https://www.google.com/search?q=matlab"
      target="_blank"
    >
      <img
        alt="no image"
        src="matlab.jpeg"
        style={
          Object {
            "height": "35px",
            "width": "35px",
          }
        }
      />
    </a>
  </div>
  <div
    className="tool-image col"
    id="ResearchAssociateatBrandeisUniversity-tool1"
    style={
      Object {
        "backgroundColor": "white",
        "maxWidth": "50px",
      }
    }
  >
    <a
      alt="link to dna"
      href="https://www.google.com/search?q=dna"
      target="_blank"
    >
      <img
        alt="no image"
        src="dna.png"
        style={
          Object {
            "height": "35px",
            "width": "35px",
          }
        }
      />
    </a>
  </div>
  <div
    className="tool-image col"
    id="ResearchAssociateatBrandeisUniversity-tool2"
    style={
      Object {
        "backgroundColor": "white",
        "maxWidth": "50px",
      }
    }
  >
    <a
      alt="link to microscope"
      href="https://www.google.com/search?q=microscope"
      target="_blank"
    >
      <img
        alt="no image"
        src="microscope.png"
        style={
          Object {
            "height": "35px",
            "width": "35px",
          }
        }
      />
    </a>
  </div>
</div>
`;
